streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java

```diff
@@ -1063,9 +1071,13 @@ private Map<TaskId, Set<TopicPartition>> buildStandbyTaskMap(final String consum
         }
 
         for (final TaskId task : revokedTasks) {
-            if (allStatefulTasks.contains(task)) {
```

Comment:
In case of rocksdb state store, will this add standby and restore state so that when active is assigned, there's more state? So it doesn't need to be a previous standby? Or this is not the intention of this code?

Another noob question, how does standby work with in-memory state store? Is it also reading from changelog topic? If so, what's the problem of losing state for in-memory state store as it can be restored from changelog topic?

Reply:
Not sure I fully understand the change here either. Would appreciate some more explanations --- specifically, could you give a simple example in the PR description?

Reply:
> Another noob question, how does standby work with in-memory state store? Is it also reading from changelog topic?

Yes.

>  If so, what's the problem of losing state for in-memory state store as it can be restored from changelog topic?

The problem is, that we would have to restore from the changelog, and we want to avoid this restore. In the end, there is a task that as the full state in-memory, and we would close the task, drop the state, and re-create the same task and restore. We basically want to avoid the "close and re-create" part. (For RocksDB it's not an issue to close a task, because when we re-open it, the state is still on disk and restoring from the changelog won't be necessary).

Reply:
> Not sure I fully understand the change here either. Would appreciate some more explanations --- specifically, could you give a simple example in the PR description?

Also not 100% sure -- I left another comment -- however I believe, we current assign standbies out of think air, even into client that should not have them, but we only want these standbies for certain client that had such a task in the previous assignment. But details are also a little bit unclear to me.

Reply:
Left a comment with an example of how this is supposed to work, and where it doesn't quite work as intended (ie the case being addressed by this fix). Hope this helps